examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,347 @@
+# (event location), and depth (if these values are not included in the argument passed
+# to ``spec``). Additionally the ``convention`` parameter is required if ``spec`` is
+# an 1-D or 2-D numpy array; for the input types dictionary and ``pandas.Dataframe``,
+# the focal mechanism convention is automatically determined from dictionary keys or
+# :class:`pandas.DataFrame` column names. The ``scale`` parameter controls the radius
+# of the beachball. By default, the value defines the size for a magnitude of 5 (i.e.,
+# a scalar seismic moment of M0 = 4.0E23 dyn cm). Appand ``"+l"`` to force the radius
+# to be proportional to the seismic moment.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(spec=mt_single, scale="1c", longitude=0, latitude=0, depth=0)
+
+fig.show()
+
+
+# %%
+# Plotting the components of a seismic moment tensor
+# --------------------------------------------------
+#
+# A moment tensor can be decomposed into isotropic and deviatoric parts. The deviatoric
+# part can be further decomposed into multiple parts (e.g., a double couple (DC) and a
+# compensated linear vector dipole (CLVD)). Use the ``component`` parameter to specify
+# the component you want to plot.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+longitude = -3
+for component in ["full", "dc", "deviatoric"]:
+    fig.meca(
+        spec=mt_single,
+        scale="1c",
+        longitude=longitude,
+        latitude=0,
+        depth=0,
+        component=component,
+    )
+    longitude = longitude + 3
+
+fig.show()
+
+
+# %%
+# Filling the quadrants
+# ---------------------
+#
+# Use the parameters ``compressionfill`` and ``extensionfill`` to fill the quadrants
+# with different colors or patterns. Regarding patterns see the gallery example
+# :doc:`Bit and hachure patterns </gallery/symbols/patterns>` and the Technical
+# Reference :doc:`Bit and hachure patterns </techref/patterns>`.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=mt_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    compressionfill="darkorange",
+    extensionfill="cornsilk",
+)
+
+fig.meca(
+    spec=mt_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="p8",
+    extensionfill="p31",
+    outline=True,
+)
+
+fig.show()
+
+
+# %%
+# Adjusting the outlines
+# ----------------------
+#
+# Use the parameters ``pen`` and ``outline`` for adjusting the circumference of the
+# beachball or all lines (circumference of the beachball and both nodal planes).
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    # Use a 1-point thick, darkorange and solid line
+    pen="1p,darkorange,solid",
```

Comment:
The default line style is `solid`, so no need to specify it here:
```suggestion
    pen="1p,darkorange",
```